google-http-client/src/main/java/com/google/api/client/http/HttpTransport.java

```diff
@@ -158,4 +158,13 @@ public boolean isMtls() {
    * @since 1.4
    */
   public void shutdown() throws IOException {}
+
+  /**
+   * Default implementation of whether the http transport is shutdown or not.
+   *
+   * @since 1.4
+   */
+  public boolean isShutdown() {
```

Comment:
It would be better to return false in the default implementation and override it to true in the child classes, similar to how `isMtls()` was implemented above. Or throw `UnsupportedOperationException()`, similar to what we did for [GAPIC stubs](https://github.com/googleapis/google-cloud-java/blob/main/java-asset/google-cloud-asset/src/main/java/com/google/cloud/asset/v1/stub/AssetServiceStub.java#L95).

Reply:
Changed to `false`.

Reply:
We would need to override it to `true` in NetHttpTransport if the default is `false`.
However, given it more thought, I think it makes more sense to return `true` as default, as you initially proposed. Because if a class extends this abstract class and don't override anything, `shutdown()` and `isShutdown()` should have consistent behavior, meaning that an no-op `shutdown()` should be accompanied with `isShutdown()` equal to `true`.

Reply:
Sure, I changed it back to `true`.